File: Examples/CSharp/AsposePDF/Forms/DynamicXFAToAcroForm.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.InteractiveFeatures.Forms;

namespace CSharp.AsposePdf.Forms
{
    public class DynamicXFAToAcroForm
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_Forms();

            // load dynamic XFA form
            Document document = new Document(dataDir + "Dynamic_XFA.pdf");

            // set the form fields type as standard AcroForm
            document.Form.Type = FormType.Standard;

            // save the resultant PDF
            document.Save(dataDir + "Standard_AcroForm.pdf");

        }
    }
}

File: Examples/VisualBasic/AsposePDF/Forms/DynamicXFAToAcroForm.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf
Imports Aspose.Pdf.Forms
Namespace AsposePDF.Forms
    Public Class DynamicXFAToAcroForm
        Public Shared Sub Run()
            ' ExStart:DynamicXFAToAcroForm
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_Forms()

            ' Load dynamic XFA form
            Dim document As New Document(dataDir & Convert.ToString("DynamicXFAToAcroForm.pdf"))

            ' Set the form fields type as standard AcroForm
            document.Form.Type = FormType.Standard

            dataDir = dataDir & Convert.ToString("Standard_AcroForm_out_.pdf")
            ' Save the resultant PDF
            document.Save(dataDir)
            ' ExEnd:DynamicXFAToAcroForm
            Console.WriteLine(Convert.ToString(vbLf & "Dynamic XFA form converted to standard AcroForm successfully." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace
